### PR TITLE
[Not Ready Yet]Refactor Layer Normalization.

### PR DESCRIPTION
**Pull request by tushushu**
### Checklist

- [ ] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

[Feat Req: Instance Normalization](https://github.com/tracel-ai/burn/issues/1249)
[Implement Instance Normalization](https://github.com/tracel-ai/burn/pull/1321)

### Changes
Refactoring the Layer Normalization based on GroupNorm, since when number of group = 1, it's equivalent to `Layernorm`. For the details, see the discussion in this [issue](https://github.com/tracel-ai/burn/issues/1249).

### Testing
Using previous unit tests.

### CODE REVIEW

**Comment by tushushu** [comment]
It seems that the way burn initialize Layer Normalization is different from Pytorch and Tensorflow, and failed to pass the UT. Close the PR for now.

**Comment by tushushu** [comment]
It seems that the way burn initialize Layer Normalization is different from Pytorch and Tensorflow, and failed to pass the UT. Close the PR for now.
